Small issue when compiling for arm64

There's a Werror that happens when compiling for arm64. It happens due to this definition.

https://github.com/SegmentLinking/TrackLooper/blob/9f96deeb3ee7b4d99fe2f8175417e50d62f6a8a0/SDL/Kernels.h#L421

Here's the Werror that it shows.

```
Kernels.h:427:30: error: comparison is always false due to limited range of data type [-Werror=type-limits]
  427 |           else if (quad_diff < 0)
      |                    ~~~~~~~~~~^~~
```

We should change it to a `int8_t` since when it's included in CMSSW it will need to be compiled for arm64.

There is also another issue with the rooutil makefile, but we can just ignore that. I'm just trying to run things locally on my laptop since it might be a bit faster to debug. I'll report if I find any other issues with arm64.

--- a/SDL/Kernels.h
+++ b/SDL/Kernels.h
@@ -418,7 +418,7 @@ namespace SDL {
           if (secondpass && (!segmentsInGPU.isQuad[jx] || (segmentsInGPU.isDup[jx] & 1)))
             continue;
 
-          char quad_diff = segmentsInGPU.isQuad[ix] - segmentsInGPU.isQuad[jx];
+          int8_t quad_diff = segmentsInGPU.isQuad[ix] - segmentsInGPU.isQuad[jx];
           float score_diff = segmentsInGPU.score[ix] - segmentsInGPU.score[jx];
           // Always keep quads over trips. If they are the same, we want the object with better score
           int idxToRemove;

--- a/code/rooutil/Makefile.arch
+++ b/code/rooutil/Makefile.arch
@@ -389,6 +389,15 @@ LDFLAGS       = -O
 SOFLAGS       = -shared
 endif
 
+ifeq ($(ARCH),linuxarm64)
+# ARM Linux with egcs
+CXX           = g++
+CXXFLAGS      = -O -Wall -fPIC -Wshadow -Woverloaded-virtual
+LD            = g++
+LDFLAGS       = -O
+SOFLAGS       = -shared
+endif
+
 ifeq ($(ARCH),mklinux)
 # MkLinux with libc5
 CXX           = g++

--- a/setup.sh
+++ b/setup.sh
@@ -6,8 +6,11 @@
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 source $DIR/code/rooutil/thisrooutil.sh
 
+ARCH=$(uname -m)
 if [[ $(hostname) == *lnx4555* ]]; then
   export SCRAM_ARCH=el9_amd64_gcc12
+elif [[ $ARCH == "aarch64" || $ARCH == "arm64" ]]; then
+  export SCRAM_ARCH=el9_aarch64_gcc12
 else
   export SCRAM_ARCH=el8_amd64_gcc12
 fi

